x-pack/plugin/fleet/src/main/java/org/elasticsearch/xpack/fleet/action/GetGlobalCheckpointsAction.java

```diff
@@ -0,0 +1,242 @@
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.util.concurrent.AtomicArray;
+import org.elasticsearch.common.util.concurrent.CountDown;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.IndexNotFoundException;
+import org.elasticsearch.index.seqno.SequenceNumbers;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+
+import java.io.IOException;
+
+public class GetGlobalCheckpointsAction extends ActionType<GetGlobalCheckpointsAction.Response> {
+
+    public static final GetGlobalCheckpointsAction INSTANCE = new GetGlobalCheckpointsAction();
+    public static final String NAME = "indices:monitor/fleet/global_checkpoints";
+
+    private GetGlobalCheckpointsAction() {
+        super(NAME, GetGlobalCheckpointsAction.Response::new);
+    }
+
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private final boolean timedOut;
+        private final long[] globalCheckpoints;
+
+        public Response(long[] globalCheckpoints) {
+            this.globalCheckpoints = globalCheckpoints;
+            this.timedOut = false;
+        }
+
+        public Response(boolean timedOut) {
+            this.timedOut = timedOut;
+            this.globalCheckpoints = new long[0];
+        }
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            timedOut = in.readBoolean();
+            globalCheckpoints = in.readLongArray();
+        }
+
+        public long[] globalCheckpoints() {
+            return globalCheckpoints;
+        }
+
+        public boolean timedOut() {
+            return timedOut;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            throw new AssertionError("GetGlobalCheckpointsAction should not be sent over the wire.");
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+            builder.startObject();
+            if (timedOut) {
```

Comment:
I would find it more client-friendly to respond with both fields always. That way they can check the `timed_out` value. This is similar to what cluster-health does.